### PR TITLE
v5.0.0

### PR DESCRIPTION
**Pull request by latotty**
Upgrade/ts4 support
added strict rules

### CODE REVIEW

File: typescript-recommended.js [hunk 1]
```diff
@@ -10,22 +10,97 @@ module.exports = {
...
+        format: ['camelCase'],
+        leadingUnderscore: 'allow',
+      },
+
+      { selector: 'memberLike', format: ['camelCase'] },
+      {
+        selector: 'memberLike',
+        modifiers: ['private'],
+        format: ['camelCase'],
+        leadingUnderscore: 'require',
```
Comment (kari0003) [inline typescript-recommended.js]: Can we change it to 'allow'?
In my experience leading underscore is useful in some cases (eg setters, getters) but obstructs readability in other cases. So i would rather have the choice to use it or not depending on my usecase.

**Review by leetm4n — APPROVED** [review]

File: typescript-recommended.js [hunk 2]
```diff
@@ -10,22 +10,97 @@ module.exports = {
...
+      },
+
+      { selector: 'typeLike', format: ['PascalCase'] },
+      { selector: 'typeParameter', format: ['PascalCase'], prefix: ['T'] },
+
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: { regex: '^I[A-Z]', match: false },
```
Comment (kari0003) [inline typescript-recommended.js]: What is the reasoning behind having `T` for typeLike prefix, but forbidding `I` for interface prefix?
I'd rather we forbid both, as they do not produce spectacular code :P

File: typescript-recommended.js (hunk 1 above)
Comment (latotty) [reply typescript-recommended.js]: Nice catch, I moved this config partly from my last project, so there may be some issues 😄

File: typescript-recommended.js (hunk 2 above)
Comment (latotty) [reply typescript-recommended.js]: Only the typeParameter must have T prefix, but I added a forbidding regex to the typelike also, thanks!

**Review by leetm4n — APPROVED** [review]

**Review by afgTheCat — APPROVED** [review]

File: typescript-recommended.js [hunk 3]
```diff
@@ -10,22 +10,100 @@ module.exports = {
...
+
+      {
+        selector: 'typeLike',
+        format: ['PascalCase'],
+        custom: { regex: '^[IT][A-Z]', match: false },
+      },
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: { regex: '^[IT][A-Z]', match: false },
```
Comment (Lacka90) [inline typescript-recommended.js:85]: maybe enough prefix: I instead of regex

**Review by Lacka90 — APPROVED** [review]

**Pull request merged**